### PR TITLE
feat(Core/DBLayer): move DatabaseWorkerPool into it's own translation unit

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
- Move DatabaseWorkerPool into its own translation unit
- Part for dynamic linking

PR based TC commit https://github.com/TrinityCore/TrinityCore/commit/09fa0ab46a487ea85d475ac0b3495724c170fe1a

##### TESTS PERFORMED:
- Build and run in Windows at VS 16.3.8

##### HOW TO TEST THE CHANGES:
- This no need test, only build in Travis

##### KNOWN ISSUES AND TODO LIST:
- [ ] Travis build

##### Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/common/Database/DatabaseWorkerPool.cpp [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+        GetDatabaseName(), async_threads, synch_threads);
+
+    //! Open asynchronous connections (delayed operations)
+    _connections[IDX_ASYNC].resize(async_threads);
+    for (uint8 i = 0; i < async_threads; ++i)
+    {
+        T* t = new T(_queue, _connectionInfo);
+        res &= t->Open();
+        if (res) // only check mysql version if connection is valid
+            WPFatal(mysql_get_server_version(t->GetHandle()) >= MIN_MYSQL_SERVER_VERSION, "AzerothCore does not support MySQL versions below 5.1");
```
Comment (Viste) [inline src/common/Database/DatabaseWorkerPool.cpp]: 5.6

File: src/common/Database/DatabaseWorkerPool.cpp [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+template <class T>
+DatabaseWorkerPool<T>::DatabaseWorkerPool() :
+_mqueue(new ACE_Message_Queue<ACE_SYNCH>(2*1024*1024, 2*1024*1024)),
+_queue(new ACE_Activation_Queue(_mqueue))
+{
+    memset(_connectionCount, 0, sizeof(_connectionCount));
+    _connections.resize(IDX_SIZE);
+
+    WPFatal(mysql_thread_safe(), "Used MySQL library isn't thread-safe.");
+    WPFatal(mysql_get_client_version() >= MIN_MYSQL_CLIENT_VERSION, "AzerothCore does not support MySQL versions below 5.1");
```
Comment (Viste) [inline src/common/Database/DatabaseWorkerPool.cpp]: 5.6

Comment (Winfidonarleyan) [reply src/common/Database/DatabaseWorkerPool.cpp]: Ah yes, thanks for review

**Comment by Winfidonarleyan** [comment]
@Yehonal Basically, I want to reduce the change between TC and AC

**Comment by BarbzYHOOL** [comment]
no idea what this does but if yehonal approves, then it's good xD

**Pull request merged**